src/main/scala/mesosphere/marathon/state/AppDefinition.scala

```diff
@@ -107,7 +107,13 @@ case class AppDefinition(
   /**
     * @return the disk resources required for volumes
     */
-  def diskForVolumes: Double = persistentVolumes.map(_.persistent.size).sum.toDouble
+  def diskForVolumes: Double = persistentVolumes.map(
+    //TODO(jdef) probably only want to consider "agent" providerName here
```

Comment:
Yes, as for we only need to reserve disk for volumes space that's provided by the agent itself